### PR TITLE
Removed github requirement

### LINKED ISSUE #197: Remove github.com requirement
Currently, there is a GitHub SDK within the code. If there is no `.sail/Dockerfile`, the base `codercom/ubuntu-dev` image should be used.

### PR DESCRIPTION
**Pull request by roberthmiller**
Closes #197

### CODE REVIEW

**Comment by roberthmiller** [comment]
@deansheather Can you have a look at this?

File: repo.go [hunk 1]
```diff
@@ -118,6 +137,10 @@ func (r repo) language() string {
 		return ""
 	}
 
+	if ok := r.isGithubRemote(); !ok {
```
Comment (deansheather) [inline repo.go]: Should be `if !r.isGithubRemote() {`

Comment (deansheather) [reply repo.go]: Also this block should be moved to the top of `language()` before `orgRepo = ...`.

File: repo.go [hunk 2]
```diff
@@ -110,6 +111,24 @@ func parseRepo(defaultSchema, defaultHost, defaultOrganization, name string) (re
 	return r, nil
 }
 
+func (r repo) isGithubRemote() (ok bool) {
+	cmd := xexec.Fmt("git remote -v")
```
Comment (deansheather) [inline repo.go]: Shouldn't be using `xexec.Fmt` here since you're not using a format string.

Comment (deansheather) [reply repo.go]: This won't work unless the current working directory is inside the repository, which isn't guaranteed. You can tell git to use a different path by giving it the `-C` parameter. Maybe `xexec.Fmt` is a good idea after all.

File: repo.go [hunk 3]
```diff
@@ -110,6 +111,24 @@ func parseRepo(defaultSchema, defaultHost, defaultOrganization, name string) (re
 	return r, nil
 }
 
+func (r repo) isGithubRemote() (ok bool) {
```
Comment (deansheather) [inline repo.go]: Should be called `isGitHubRemote` (capital H).

Comment (deansheather) [reply repo.go]: `(ok bool)` should just be `bool`.

File: repo.go [hunk 4]
```diff
@@ -110,6 +111,24 @@ func parseRepo(defaultSchema, defaultHost, defaultOrganization, name string) (re
...
+	cmd := xexec.Fmt("git remote -v")
+	out, err := cmd.Output()
+	if err != nil {
+		flog.Error("Unable to check repo remotes")
```
Comment (deansheather) [inline repo.go]: `flog.Error("unable to check Git remotes: %s", out)`

File: repo.go [hunk 5]
```diff
@@ -110,6 +111,24 @@ func parseRepo(defaultSchema, defaultHost, defaultOrganization, name string) (re
 	return r, nil
 }
 
+func (r repo) isGithubRemote() (ok bool) {
+	cmd := xexec.Fmt("git remote -v")
+	out, err := cmd.Output()
```
Comment (deansheather) [inline repo.go]: Use [`cmd.CombinedOutput`](https://golang.org/pkg/os/exec/#Cmd.CombinedOutput) so we can use this in the error as well.

File: repo.go (hunk 3 above)
Comment (deansheather) [reply repo.go]: Also, instead of logging errors, this function should return them.

File: repo.go [hunk 6]
```diff
@@ -110,6 +111,24 @@ func parseRepo(defaultSchema, defaultHost, defaultOrganization, name string) (re
...
+	if err != nil {
+		flog.Error("Unable to check repo remotes")
+		return false
+	}
+
+	o := strings.Split(string(out), "\n")
```
Comment (deansheather) [inline repo.go]: This won't work, as `git remote -v` is prefixed with the name of each remote.

File: repo.go [hunk 7]
```diff
@@ -110,6 +111,24 @@ func parseRepo(defaultSchema, defaultHost, defaultOrganization, name string) (re
...
+
+	for _, url := range o {
+		if strings.HasPrefix(url, "https://github.com") || strings.HasPrefix(url, "git@github.com") {
```
Comment (deansheather) [inline repo.go]: We should at least know the `username/repo` of the repository, so I think we should check that the URL matches exactly against what we expect it to be. For example, for Sail it should be `https://github.com/cdr/sail.git` or `git@github.com:cdr/sail.git`.

**Comment by roberthmiller** [comment]
Do you think that just changing the `HasPrefix` method should simply be changed to `Contains`?

File: repo.go (hunk 1 above)
Comment (roberthmiller) [reply repo.go]: my bad, was tired when putting this together 😄

Comment (roberthmiller) [reply repo.go]: Also, I would not be able to move this if I included a specific check for the org repo 😛. I think it should stay where it is for now. Once I update the PR, you can have another look.

**Comment by roberthmiller** [comment]
@deansheather, I am unsure how I would get the path of the repo from the current code... Any ideas?

**Comment by deansheather** [comment]
Actually, I just had another look at repo.go. Why can't we just use `r.Host` to check if the repository is on GitHub? That would simplify this PR significantly.

**Comment by deansheather** [comment]
https://gist.github.com/deansheather/af960f705d938f4ef67fa4332d711202

**Comment by coadler** [comment]
Getting the origin url is probably the best bet.

**Comment by roberthmiller** [comment]
I think that would work for sure. Will change this when I have a moment

**Comment by deansheather** [comment]
If we're getting the origin URL from `git`, I'd suggest something more like: https://gist.github.com/deansheather/d83477514fa0b54cfa04e1c0f4f355c4

@coadler

**Comment by coadler** [comment]
The diff gist is a bit hard to parse, but `url.Parse` would definitely be better here, and your diff seems to handle the `:` correctly.

**Comment by coadler** [comment]
Also, I may have misled you. Reread through the comments and my

> Getting the origin url is probably the best bet.

comment was in response to the PR comment you pinged me in and I didn't notice the comment before it pointing out `r.Host`. 

`r.Host` should solve this problem, my apologies.

**Comment by deansheather** [comment]
Alright, I'm going to apply the first diff then.

**Comment by deansheather** [comment]
One issue I just found with `r.Host` is that it's not correct if you do something like:
```
sail run gitlab.com/deansheather/some-repo => r.Host is gitlab.com
sail run deansheather/some-repo => r.Host is github.com even though it's already cloned from another host
```

**Comment by coadler** [comment]
Unlucky, didn’t think about that

**Comment by deansheather** [comment]
I think the best way to fix this is probably to add the `git remote get-url
origin` check to `parseRepo()` if the repo is already on disk. Problem with
that, though, is that parseRepo doesn't know where to look for the
repository, so we'll have to add even more params.
